sql: gate Geospatial features under version gate

otan opened the pull request:
Prevent the creation of Geospatial types until a version upgrade is
complete. Re-using the 19.2-20.1 mixed cluster tests for now to test
this.

Resolves #47242.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47800)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47800#-:-M5XRtky5M8xU8cDdblA:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/create_table.go, line 120 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47800#-M5XRcYbDQDkNqRfJkmh:-M5XRcYbDQDkNqRfJkmi:b-valo5y) ([raw file](https://github.com/cockroachdb/cockroach/blob/1773029d388a51c66dfed34aea57011e8a8160d5/pkg/sql/create_table.go#L120)):*
> ```Go
> 	types.TimeTZFamily:    clusterversion.VersionTimeTZType,
> 	types.GeographyFamily: clusterversion.VersionGeospatialType,
> 	types.GeometryFamily:  clusterversion.VersionGeospatialType,
> ```

I am assuming this means I don't need to do any version check for geospatial indexes, since they will appear in the same version and the geospatial type is already gated. Correct?


<!-- Sent from Reviewable.io -->

otan commented:
___
*[pkg/sql/create_table.go, line 120 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47800#-M5XRcYbDQDkNqRfJkmh:-M5XW5HP269gU1bGlj4a:bppphvo) ([raw file](https://github.com/cockroachdb/cockroach/blob/1773029d388a51c66dfed34aea57011e8a8160d5/pkg/sql/create_table.go#L120)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

I am assuming this means I don't need to do any version check for geospatial indexes, since they will appear in the same version and the geospatial type is already gated. Correct?
</blockquote></details>

Yeah, you can to be double sure but either way


<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola

otan commented:
bors r=sumeerbhola

yuzefovich commented:
Sorry for jumping in, but I believe this PR doesn't pass the lint check (https://teamcity.cockroachdb.com/viewLog.html?buildId=1893705&buildTypeId=Cockroach_UnitTests)

bors r-

otan commented:
yep! all those CI failures yesterday hid the problem

otan committed:
sql: gate Geospatial features under version gate

Prevent the creation of Geospatial types until a version upgrade is
complete. Re-using the 19.2-20.1 mixed cluster tests for now to test
this.

Release note: None

otan commented:
bors r=sumeerbhola
